src/main/java/com/google/api/gax/grpc/ApiCallSettingsTyped.java

```diff
@@ -1,40 +1,92 @@
+ * call settings classes inherit from.
  */
 class ApiCallSettingsTyped<RequestT, ResponseT> extends ApiCallSettings {
-  protected MethodDescriptor<RequestT, ResponseT> methodDescriptor;
 
-  protected FutureCallable<RequestT, ResponseT> createFutureCallable() {
-    ClientCallFactory<RequestT, ResponseT> clientCallFactory =
-        new DescriptorClientCallFactory<>(methodDescriptor);
-    return new DirectCallable<>(clientCallFactory);
+  private final MethodDescriptor<RequestT, ResponseT> methodDescriptor;
+
+  public MethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
+    return methodDescriptor;
   }
 
-  ApiCallSettingsTyped(Builder<RequestT, ResponseT> builder) {
-    super(builder);
-    this.methodDescriptor = builder.getDescriptor();
+  public ApiCallSettingsTyped(Set<Status.Code> retryableCodes,
+                              RetrySettings retrySettings,
+                              MethodDescriptor<RequestT, ResponseT> methodDescriptor) {
+    super(retryableCodes, retrySettings);
+    this.methodDescriptor = methodDescriptor;
   }
 
   public static class Builder<RequestT, ResponseT>
       extends ApiCallSettings.Builder {
-    protected MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor;
+    private MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor;
 
     public Builder(MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor) {
       this.grpcMethodDescriptor = grpcMethodDescriptor;
     }
 
-    public MethodDescriptor<RequestT, ResponseT> getDescriptor() {
+    public MethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
       return grpcMethodDescriptor;
     }
 
     @Override
     public ApiCallSettingsTyped<RequestT, ResponseT> build() {
-      return new ApiCallSettingsTyped<RequestT, ResponseT>(this);
+      return new ApiCallSettingsTyped<RequestT, ResponseT>(
+          getRetryableCodes(), getRetrySettingsBuilder().build(), grpcMethodDescriptor);
+    }
+  }
+
+  protected FutureCallable<RequestT, ResponseT> createFutureCallable() {
+    ClientCallFactory<RequestT, ResponseT> clientCallFactory =
+        new DescriptorClientCallFactory<>(methodDescriptor);
+    return new DirectCallable<>(clientCallFactory);
+  }
+
+  protected ApiCallable<RequestT, ResponseT> setupCallable(
+      ApiCallable<RequestT, ResponseT> callable,
+      ServiceApiSettings.Builder serviceSettingsBuilder) throws IOException {
+    ManagedChannel channel = serviceSettingsBuilder.getOrBuildChannel();
+    ScheduledExecutorService executor = serviceSettingsBuilder.getOrBuildExecutor();
+
+    if (getRetryableCodes() != null) {
+      callable = callable.retryableOn(ImmutableSet.copyOf(getRetryableCodes()));
+    }
+
+    if (getRetrySettings() != null) {
+      callable = callable.retrying(getRetrySettings(), executor);
+    }
+    return callable.bind(channel);
+  }
+
+  protected <ResourceT> ApiCallable<RequestT, Iterable<ResourceT>> setupIterableCallable(
```

Comment:
This is actually not necessary. In PageStreamingCallSettings, you should be using ApiCallable.pageStreaming to convert an ApiCallable<RequestT, ResponseT> to ApiCallable<RequestT, Iterable<ResourceT>>.